pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/TransactionAggregator.java

```diff
@@ -67,7 +68,11 @@ public static void generate(PulsarService pulsar, SimpleTextOutputStream stream,
                                     generateManageLedgerStats(managedLedger,
                                             stream, cluster, namespace, name, subscription.getName());
                                 } catch (Exception e) {
-                                    log.warn("Transaction pending ack generate managedLedgerStats fail!", e);
+                                    if (e.getCause() instanceof BrokerServiceException.NotAllowedException) {
```

Comment:
Instead of throwing exception which has a high cost (since it has to generate the entire stack trace), we could change the method to return `CompletableFuture<Optional<ManagedLedger>>` instead

Reply:
yeah, good idea. done. Thanks a lot.